### DISCUSSION TITLE
nginx error logs

### DISCUSSION THREAD

**Discussion by peterreji** [martin-helmich/prometheus-nginxlog-exporter#185] [General]
Can we parse Nginx error logs too using this exporter?

**Comment by martin-helmich** [comment]
I haven't done this myself, but in theory, you should be able to, yes. It was hard to find a definitive source on the actual format used for the NGINX error log -- all I could find was a [StackOverflow post](https://stackoverflow.com/a/26125951/1995300), according to which the error log follows the following log format:

> YYYY/MM/DD HH:MM:SS [LEVEL] PID#TID: *CID MESSAGE

In that case, the following configuration snippet might do the trick (untested -- let me know if and how it works out 😉):

```hcl
namespace "nginx" {
  format = "$date $time [$level] $pid#$tid: *$cid $message"

  relabel "level" {
    from = "level"
  }
  // ...
}
```

I'm not really sure which values are sensible to be used as labels -- apart from `$level`, I don't see any good label candidates in there, but `$level` alone would already allow metrics like "error rate per second and error level".
